x/staking/src/keeper/validators_and_total_power.rs

```diff
@@ -96,6 +98,60 @@ impl<
         store.delete(&validator.key_by_power_index_key(power_reduction))
     }
 
+    /// get the last validator set
+    pub fn last_validators_by_addr<DB: Database, CTX: QueryableContext<DB, SK>>(
+        &self,
+        ctx: &CTX,
+    ) -> HashMap<String, Vec<u8>> {
+        let mut last = HashMap::new();
+        let store = ctx.kv_store(&self.store_key);
+        let store = store.prefix_store(LAST_VALIDATOR_POWER_KEY);
+        for (k, v) in store.range(..).to_infallible_iter() {
```

Comment:
@NYBACHOK this is why I think we should get rid of `to_infallible_iter`.

Reply:
You mean @dmvict overuses it? I don't think absolutely good idea...

Reply:
Maybe something like `#[deprecated]` will do the trick to better inform user